### ISSUE TITLE
Document doesn't exist when parse callback fires

### ISSUE THREAD

**Issue by nrw** [aheckmann/gridform#1]
In the `form.parse` callback, the new file's document is only available after a delay. I don't think this the intended behavior. Am I misusing `gridform`?

``` javascript
// Requires omitted. I don't think they're the
// problem as every other operation using
// node-mongo-native behaves as expected

app.post('/', function(req, res) {
  var form = gridform({
    db: db,
    mongo: mongo
  });

  form.parse(req, function(err, fields, files) {
    var file = files.upload;

    // version 1: this works
    setTimeout(function() {
      db.collection('fs.files', function(err, coll) {
        coll.findOne({_id: file.id}, function(err, doc) {
          // doc exists
        });
      });
    }, 30);

    // version 2: this doesn't work
    db.collection('fs.files', function(err, coll) {
      coll.findOne({_id: file.id}, function(err, doc) {
        // doc is null
      });
    });

  });
});
```

**Comment by aheckmann** [comment]
not misusing. we'll need to review if we're listening to the proper event to trigger completion.

**Comment by nrw** [comment]
it looks like the write stream `end` event is the trigger. is there a `close` event we can wait for?

**Comment by aheckmann** [comment]
fixed in 0.1.2
